multiplatform/greeter/shared/build.gradle.kts

```
@@ -0,0 +1,74 @@
+plugins {
+  alias(libs.plugins.kotlin.multiplatform)
+  alias(libs.plugins.android.library)
+  alias(libs.plugins.ksp)
+}
+
+kotlin {
+  android()
+
+  listOf(
+    iosX64(),
+    iosArm64(),
+    iosSimulatorArm64()
+  ).forEach {
+    it.binaries.framework {
+      baseName = "shared"
+    }
+  }
```

You can use the ios target to simplify this

```suggestion
  ios {
    binaries.framework {
      baseName = "shared"
    }
  }
```

---

I tried to touch as little as possible on the Gradle setup to keep it as similar as possible to a new KMM project (from the [KMM plugin](https://plugins.jetbrains.com/plugin/14936-kotlin-multiplatform-mobile)), so it's easy for anyone looking at it.

There's an extra reason to keep this as it currently is, though: the `ios` target shortcut will most likely be deprecated in the future and [Kotlin 1.8.20 already introduced a new way to deal with iOS targets](https://kotlinlang.org/docs/whatsnew1820.html#new-approach-to-source-set-hierarchy): 

>Kotlin 1.8.20 offers a new way of setting up source set hierarchy in your multiplatform projects − the default target hierarchy. The new approach is intended to replace target shortcuts like ios, which have their [design flaws](https://kotlinlang.org/docs/whatsnew1820.html#why-replace-shortcuts).

In the future, instead of using the `ios` shortcut, we'll have something like this instead:

```
targetHierarchy.default()

iosX64()
iosArm64()
iosSimulatorArm64()
```

So I'd prefer not to use the shortcut, and as soon as there's a stable Compose compiler version supporting Kotlin 1.8.20, I can update the sample with this change.

---

Oh thanks for the heads up! Was unaware of this development. You are right that the current way is closer to how it looks like it should work in the future.